common/djangoapps/course_modes/helpers.py

```diff
@@ -79,7 +80,10 @@ def _enrollment_mode_display(enrollment_mode, verification_status, course_id):
     course_mode_slugs = [mode.slug for mode in CourseMode.modes_for_course(course_id)]
 
     if enrollment_mode == CourseMode.VERIFIED:
-        if verification_status in [VERIFY_STATUS_NEED_TO_VERIFY, VERIFY_STATUS_SUBMITTED, VERIFY_STATUS_APPROVED]:
+        if (
+            is_integrity_signature_enabled(course_id)
```

Comment:
I don't really like that this has to be duplicated twice, but if I remove the check for verification status value here, the expected values aren't returned when the waffle flag is off.

Reply:
it's not quite the same check but covers some of the same ground so 🤷